Feature/clan info display v5

libertus21 opened the pull request:
Added the descriptions of the clans in 5Th sheet, along with their disciplines, curse and compulsion in the frontend

![image](https://github.com/user-attachments/assets/3bdd4e59-c297-469d-9e6e-27c3a355954f)

![image](https://github.com/user-attachments/assets/0c9d9b5f-eaab-4a6d-8c6a-814ad018981c)

libertus21 committed:
Add detailed clan information display for V5  - Add clan descriptions, banes and compulsions from V5 rulebook - Implement info dialog with responsive design

libertus21 committed:
Extract ClanInfoDialog into separate compone

Mirai-Miki commented:
I shall test it soon. Are you able to change the pull request into the [development branch](https://github.com/Mirai-Miki/Realm-of-Darkness-Bot/tree/development)? I want things to be production tested before moving them into main.

libertus21 commented:
Done the change.

libertus21 committed:
Merge branch 'development' into feature/clan-info-display-v5

Mirai-Miki committed:
Merge branch 'development' into pr/11

Mirai-Miki committed:
Refactor clan info dialog and update clan data

Replaced ClanInfoDialog with a more generic ItemInfoDialog to support displaying information for various item types, not just clans. Updated SelectionDialogue to use ItemInfoDialog and improved info button logic. Expanded and revised clan details in constants/index.js, including new fields and more comprehensive descriptions. Added 'Tithe Collector' to predator types in both frontend and backend models.
